### PR TITLE
Made the oscilloscope more efficient. Distortions reduced.

### PR DESCRIPTION
**Pull request by ragv**
1. The sampling method is now more efficient since the `logicArray` is not a part of the dictionary but a 2 Dimensional list [ `self.WID` X `self.MAX_INP` ] .
2. `xrange` function will be called instead of `range` [ only for Py 2.x ]
3. Length of the inputs `self.leninputs` will be precalculated to improve efficiency at the time of sampling.

Effect of this is that the distortion produced in the waveforms especially the ones produced in the clocks will be reduced compared to before.  [ as long as the sampling freq is set to > 2\* Nyquist freq. (preferably 5*Nyq. freq.) ]

### CODE REVIEW

File: BinPy/examples/Counters/NBitDownCounter.py [hunk 1]
```diff
@@ -6,52 +6,33 @@
 from BinPy.Gates import Connector
 from BinPy.tools.oscilloscope import Oscilloscope
 
-print("Initialize a toggle connectr for inpput in TFlipFlop")
-print("toggle = Connector(1)")
-toggle = Connector(1)
```
Comment (rajat974) [inline BinPy/examples/Counters/NBitDownCounter.py:11]: why ??

Comment (rajat974) [inline BinPy/examples/Counters/NBitDownCounter.py:11]: Example cant produce error, what is the need of changing it?

Comment (ragv) [inline BinPy/examples/Counters/NBitDownCounter.py:11]: Looks like all the examples are not working ... u added the lines after `clock.kill()` in your pr #163 .

Comment (rajat974) [inline BinPy/examples/Counters/NBitDownCounter.py:11]: #171

**Comment by debugger22** [comment]
@jayrambhia

**Pull request merged**